tests/components/wemo/test_init.py

```diff
@@ -87,3 +92,59 @@ async def test_static_config_with_invalid_host(hass):
         },
     )
     assert not setup_success
+
+
+async def test_discovery(hass, pywemo_registry):
+    """Verify that discovery dispatches devices to the platform for setup."""
+
+    def create_device(counter):
+        """Create a unique mock Motion detector device for each counter value."""
+        device = create_autospec(pywemo.Motion, instance=True)
+        device.host = f"{MOCK_HOST}_{counter}"
+        device.port = MOCK_PORT + counter
+        device.name = f"{MOCK_NAME}_{counter}"
+        device.serialnumber = f"{MOCK_SERIAL_NUMBER}_{counter}"
+        device.model_name = "Motion"
+        device.get_state.return_value = 0  # Default to Off
+        return device
+
+    pywemo_devices = [create_device(0), create_device(1)]
+    mock_stop = Mock()
+    expected_device_count = 4
+
+    def async_call_later(hass, delay, action):
+        async def async_run_action_after_platform_setup():
+            """Run the 'action' after the platform has been setup.
+
+            This forces the async_dispatcher_send logic to be tested.
+            """
+            await pywemo_registry.semaphore.acquire()  # Returns after platform setup.
+            count = len(pywemo_devices)
+            if count < expected_device_count:
+                pywemo_devices.append(create_device(count))
+                await action(dt_util.utcnow())
+
+        hass.async_create_task(async_run_action_after_platform_setup())
+        return mock_stop
+
+    # Setup the component and start discovery.
+    with patch(
+        "pywemo.discover_devices", return_value=pywemo_devices
+    ) as mock_discovery, patch(
+        "homeassistant.components.wemo.async_call_later", side_effect=async_call_later
```

Comment:
Can we take control of time and move it forward instead of patching async_call_later? We have a test helper:
https://github.com/home-assistant/core/blob/317ed418dd19dc7dcf2906ddaf3ad6b04583bf2c/tests/common.py#L287

Reply:
That cleans this code up nicely. Thanks for the pointer.